onnxruntime/core/session/inference_session_utils.cc

```diff
@@ -0,0 +1,281 @@
+  return Status::OK();
+}
+
+static Status set_inter_op_num_threads(SessionOptions& session_options,
+                                       int value,
+                                       const logging::Logger& logger) {
+  if (value < 0) {
+    LOGS(logger, ERROR) << "Unsupported value for inter_op_num_threads: " << value;
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported value for inter_op_num_threads: ", value);
+  }
+
+  LOGS(logger, INFO) << "Setting inter_op_num_threads to " << value;
+  session_options.inter_op_num_threads = value;
+  return Status::OK();
+}
+
+static Status set_execution_mode(SessionOptions& session_options,
+                                 int value,
+                                 const logging::Logger& logger) {
+  switch (value) {
+    case ExecutionMode::ORT_SEQUENTIAL:
+      LOGS(logger, INFO) << "Setting execution_mode to Sequential mode";
+      session_options.execution_mode = ExecutionMode::ORT_SEQUENTIAL;
+      return Status::OK();
+
+    case ExecutionMode::ORT_PARALLEL:
+      LOGS(logger, INFO) << "Setting execution_mode to Parallel mode";
+      session_options.execution_mode = ExecutionMode::ORT_PARALLEL;
+      return Status::OK();
+
+    default:
+      LOGS(logger, ERROR) << "Unsupported execution_mode value in ORT config: " << value;
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Unsupported execution_mode value in ORT config: ", value);
+  }
+}
+
+static Status set_graph_optimization_level(SessionOptions& session_options,
+                                           int value,
+                                           const logging::Logger& logger) {
+  switch (value) {
+    case 0:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_DISABLE_ALL";
+      session_options.graph_optimization_level = TransformerLevel::Default;
+      return Status::OK();
+
+    case 1:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_BASIC";
+      session_options.graph_optimization_level = TransformerLevel::Level1;
+      return Status::OK();
+
+    case 2:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_EXTENDED";
+      session_options.graph_optimization_level = TransformerLevel::Level2;
+      return Status::OK();
+
+    case 3:
+      LOGS(logger, INFO) << "Setting graph_optimization_level to ORT_ENABLE_ALL";
+      session_options.graph_optimization_level = TransformerLevel::Level3;
+      return Status::OK();
+
+    default:
+      LOGS(logger, ERROR) << "Unsupported graph_optimization_level value in ORT config: " << value;
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Unsupported graph_optimization_level value in ORT config: ", value);
+  }
+}
+
+static Status set_enable_profiling(SessionOptions& session_options,
+                                   int value,
+                                   const logging::Logger& logger) {
+  switch (value) {
```

Comment:
Could be shortened like if (value!=0 && value!=1) return error.

Reply:
Good idea. I made the change and also made it for 'execution_mode' which similarly has just 2 valid values.